Kickstarter-iOS/Views/Controllers/AddNewCardViewController.swift

```diff
@@ -251,9 +252,7 @@ STPPaymentCardTextFieldDelegate, MessageBannerViewControllerPresenting {
   }
 
   private func dismissAndPresentMessageBanner(with message: String) {
-    self.navigationController?.dismiss(animated: true, completion: { [weak self] in
-      self?.delegate?.presentAddCardSuccessfulBanner(message)
-    })
+    self.delegate?.addNewCardViewControllerSucceeded(with: message)
```

Comment:
Delegate should be responsible for the rest (no need to dismiss anything from here as it adds more responsibility to the controller which it shouldn't be aware of)